src/stan/mcmc/chainset.hpp

```diff
@@ -0,0 +1,504 @@
+    return Eigen::Map<Eigen::VectorXd>(quantiles.data(), quantiles.size());
+  }
+
+  /**
+   * Compute the quantile values of the specified parameter
+   * for a set of specified probabilities.
+   *
+   * Throws exception if any probability is not between 0 and 1.
+   *
+   * @param name parameter name
+   * @param probs vector of probabilities
+   * @return vector of parameter values for quantiles
+   */
+  Eigen::VectorXd quantiles(const std::string& name,
+                            const Eigen::VectorXd& probs) const {
+    return quantiles(index(name), probs);
+  }
+
+  /**
+   * Computes the split potential scale reduction (split Rhat) using rank based
+   * diagnostic for a set of per-chain draws, for bulk and tail Rhat.
+   * Based on paper https://arxiv.org/abs/1903.08008
+   *
+   * @param index parameter index
+   * @return pair (bulk_rhat, tail_rhat)
+   */
+  std::pair<double, double> split_rank_normalized_rhat(const int index) const {
+    return analyze::split_rank_normalized_rhat(samples(index));
+  }
+
+  /**
+   * Computes the split potential scale reduction (split Rhat) using rank based
+   * diagnostic for a set of per-chain draws, for bulk and tail Rhat.
+   * Based on paper https://arxiv.org/abs/1903.08008
+   *
+   * @param name parameter name
+   * @return pair (bulk_rhat, tail_rhat)
+   */
+  std::pair<double, double> split_rank_normalized_rhat(
+      const std::string& name) const {
+    return split_rank_normalized_rhat(index(name));
+  }
+
+  /**
+   * Computes the effective sample size (ESS) for the specified
+   * parameter across all chains, according to the algorithm presented in
+   * https://arxiv.org/abs/1903.08008, section 3.2 for folded (split)
+   * rank-normalized ESS for both bulk and tail ESS.
+   *
+   * @param index parameter index
+   * @return pair (bulk_ess, tail_ess)
+   */
+  std::pair<double, double> split_rank_normalized_ess(const int index) const {
+    return analyze::split_rank_normalized_ess(samples(index));
+  }
+
+  /**
+   * Computes the effective sample size (ESS) for the specified
+   * parameter across all chains, according to the algorithm presented in
+   * https://arxiv.org/abs/1903.08008, section 3.2 for folded (split)
+   * rank-normalized ESS for both bulk and tail ESS.
+   *
+   * @param name parameter name
+   * @return pair (bulk_ess, tail_ess)
+   */
+  std::pair<double, double> split_rank_normalized_ess(
+      const std::string& name) const {
+    return split_rank_normalized_ess(index(name));
+  }
+
+  /**
+   * Computes the mean Monte Carlo error estimate for the central 90% interval.
+   * See https://arxiv.org/abs/1903.08008, section 4.4.
+   * Follows implementation in the R posterior package
+   *
+   * @param index parameter index
+   * @return pair (bulk_ess, tail_ess)
+   */
+  double mcse_mean(const int index) const {
+    double ess_bulk = analyze::split_rank_normalized_ess(samples(index)).first;
+    return sd(index) / std::sqrt(ess_bulk);
```

Comment:
Question for @avehtari from above about which ESS this should use:

>  The old ESS which (called ess_basic() in posterior package) is still needed for computing MCSE for mean estimate.

?

Reply:
mcse_mean should use ess_mean, not ess_bulk
in the same way, mcse_sd should not use the rank normalized ess